src/main.cc

```diff
@@ -832,5 +832,14 @@ main(int argc, char **argv)
 	}
 	/* freeing resources */
 	tarantool_free();
+	ERROR_INJECT(ERRINJ_MAIN_MAKE_FILE_ON_RETURN, do {
+		int fd = open("tt_exit_file.txt.inprogress",
+			      O_WRONLY | O_CREAT | O_TRUNC, -1);
+		if (fd < 0)
+			break;
+		dprintf(fd, "ExitCode: %d\n", exit_code);
+		close(fd);
+		rename("tt_exit_file.txt.inprogress", "tt_exit_file.txt");
+	} while (false));
```

Comment:
AFAIU this error injection is just a way to workaround the bug https://github.com/tarantool/luatest/issues/252 in luatest. If so, it isn't worth it IMO. Let's drop it?

Actually, I wouldn't even bother with the test, since the issue is (a) pretty artificial (doesn't affect customers) and (b) detected by other tests now and then, anyway. WDYT?

Reply:
Yes, apparently it is a workaround for that luatest ticket. Didn't know about that one.

As for dropping - I would rather want to keep it, if you don't insist. At least until the luatest bug is fixed. The reasons are:
- I don't believe it will be fixed in luatest soon, really;
- Other tests can't cover it at least until the luatest bug is fixed;
- Other tests might cover is sometimes, but this test is a 100% reproducer.
- Even after luatest bug will be fixed, the test will work even without a fix on MacOS because of this: #8023. Basically, on my PC I simply can't catch this bug anyhow right now even if I would waitpid the child process and check its exit status.

Also as a bonus we can add to this exit-file additional into. At least this is what I had in mind. The thing is that at exit you can't log anything already - log is destroyed here (firstly I tried writing the exit code to logs).

You might try writing to stdout and capture it from in your test. I investigated how to do that via luatest, but it was extremely ugly if possible at all (I saw code but haven't tried it). Should be simpler if luatest will ever switch to our popen.

But you still might want to get info like how many threads were created/deleted/still run, or how much memory we didn't free, or likewise. That is why I not just create an empty file, but add here a key `ExitCode` - in future we can very easy add more keys for various stuff similarly. Or, well, drop the test when luatest is fixed and return it if needed anytime. For now I would very much like to keep it.